pennylane/measurements/shots.py

```diff
@@ -135,7 +135,7 @@ class Shots:
 
     _frozen = False
 
-    def __new__(cls, shots):
+    def __new__(cls, shots=None):
```

Comment:
Yes, we can't put it in the `Resources` object without having a default value here. This is because the `Resources` object is a frozen data class, it requires types that can be initialized without any arguments. There was a way to make it work with an alternate constructor, but after speaking with Christina we decided this makes more sense.